include/fslock.h

```diff
@@ -1,30 +1,22 @@
 #ifndef NEWSBOAT_FSLOCK_H_
 #define NEWSBOAT_FSLOCK_H_
 
+#include "fslock.rs.h"
+
 #include <string>
 #include <sys/types.h>
 
-extern "C" {
-
-	void* rs_fslock_new();
-
-	void rs_fslock_free(void* ptr);
-
-	bool rs_fslock_try_lock(void* ptr, const char* new_lock_filepath, pid_t* pid);
-
-} // extern "C"
-
 namespace newsboat {
 
 class FsLock {
 public:
 	FsLock();
 	~FsLock();
 
-	bool try_lock(const std::string& lock_file, pid_t& pid);
+	bool try_lock(const std::string& lock_file, std::int64_t& pid);
```

Comment:
I'd keep `pid_t`, and use `libc::pid_t` on the Rust side. A lot of our dependencies depend on libc crate, so we can use it quite confidently — we won't be dropping it anytime soon.

Reply:
Amended into: https://github.com/newsboat/newsboat/pull/1317/commits/682bb8d0011071abfe4aabf8970f3a0887b1f647

Reply:
It looks like only C++ side got amended? Rust still uses `i64`. I haven't tried it — is there a problem using `libc::pid_t` with cxx?

Reply:
From what I understand, cxx only supports a limited number of types in structs, arguments, and returns:
https://cxx.rs/bindings.html#built-in-bindings-reference
`libc::pid_t` is not in that list.

![image](https://user-images.githubusercontent.com/4629607/100549963-ca825180-3276-11eb-9b82-c01a309214b4.png)

Reply:
Ah. Okay then, let's keep a wrapper on the Rust side.

Reply:
Supporting stuff like this is in scope for CXX but is just not implemented yet. See https://github.com/dtolnay/cxx/issues/529 for an issue to follow.